openpype/plugins/publish/preintegrate_thumbnail_representation.py

```diff
@@ -0,0 +1,68 @@
+""" Marks thumbnail representation for integrate to DB or not.
+
+    Some hosts produce thumbnail representation, most of them do not create
+    them explicitly, but they created during extract phase.
+
+    In some cases it might be useful to override implicit setting for host/task
+
+    This plugin needs to run after extract phase, but before integrate.py as
+    thumbnail is part of review family and integrated there.
+
+    It should be better to control integration of thumbnail in one place than
+    configure it in multiple places on host implementations.
+"""
+import pyblish.api
+
+from openpype.lib.profiles_filtering import filter_profiles
+
+
+class PreIntegrateThumbnails(pyblish.api.InstancePlugin):
+    """Marks thumbnail representation for integrate to DB or not."""
+
+    label = "Should Integrate Thumbnails"
+    order = pyblish.api.IntegratorOrder - 0.1
+    families = ["review"]
+
+    integrate_profiles = {}
+
+    def process(self, instance):
+        thumbnail_repre = None
+        for repre in instance.data["representations"]:
+            if repre["name"] == "thumbnail":
+                thumbnail_repre = repre
+                break
+
+        if not thumbnail_repre:
+            return
+
+        family = instance.data["family"]
+        subset_name = instance.data["subset"]
+        host_name = instance.context.data["hostName"]
+
+        anatomy_data = instance.data["anatomyData"]
+        task = anatomy_data.get("task", {})
+
+        found_profile = filter_profiles(
+            self.integrate_profiles,
+            {
+                "hosts": host_name,
+                "tasks": task.get("name"),
```

Comment:
Note:
We have combined usage of `"tasks"` and `"task_names"` in profiles and I would say that `"task_names"` is better (as is more explicit).